extensions/amp-video-iframe/1.0/amp-video-iframe.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {BUBBLE_MESSAGE_EVENTS} from '../0.1/amp-video-iframe';
+import {VideoBaseElement} from '../../amp-video/1.0/base-element';
+import {VideoIframe} from '../../amp-video/1.0/video-iframe';
+import {createCustomEvent} from '../../../src/event-helper';
+import {dict} from '../../../src/utils/object';
+
+/** @const {string} */
+const TAG = 'amp-video-iframe';
+
+class AmpVideoIframe extends VideoBaseElement {}
+
+/**
+ * @param {!MessageEvent} e
+ */
+const onMessage = (e) => {
+  const method = e.data?.method;
+  if (method) {
+    if (method == 'getIntersection') {
+      // TODO
+      return;
+    }
+    throw new Error(`Unknown method ${method}`);
+  }
+  const event = e.data?.event;
+  if (!event) {
+    return;
+  }
+  if (event === 'analytics') {
+    // TODO
+    return;
+  }
+  if (event === 'error') {
+    // TODO
+    return;
+  }
+  if (BUBBLE_MESSAGE_EVENTS.indexOf(event) > -1) {
+    e.currentTarget.dispatchEvent(
+      createCustomEvent(window, event, /* detail */ null, {
+        bubbles: true,
+        cancelable: true,
+      })
+    );
+    return;
+  }
+};
+
+/**
+ * @param {string} method
+ * @return {method}
+ */
+const makeMethodMessage = (method) =>
+  JSON.stringify({
+    'event': 'method',
+    'method': method.toLowerCase(),
+  });
+
+AmpVideoIframe['staticProps'] = dict({
+  'component': VideoIframe,
+  'onMessage': onMessage,
+  'makeMethodMessage': makeMethodMessage,
```

Comment:
This doesn't need a specific component, but only specific `props`. 

This can be the case for 3p players like `amp-youtube` too, where we can specify its behavior only by changing props.

In a future world where CE get compiled away, for other components this can be expressed in one of two ways.

From:
```html
<amp-youtube ...></amp-youtube>
```

Into: 

A.

```js
<Youtube ... />
```

or B.

```js
<VideoIframe makeMethodMessage={YOUTUBE_MAKE_METHOD_MESSAGE} ... />
```

This is irrelevant for this PR, but it might be interesting to note.